src/File.php

```diff
@@ -803,6 +803,20 @@ public function onBeforePublish()
         }
     }
 
+    public function renameFile($newName)
```

Comment:
Under what circumstance will this method be called? I feel that in not actually renaming the file until a write we've broken the contract of the interface.

An alternative approach would be to call `$this->write()` as part of this call, or to trigger a single-field write if we didn't want  to do that.

However, the implications of this depend on where this method will be called from.